internal/stringutil/split_test.go

```diff
@@ -0,0 +1,72 @@
+package stringutil
+
+import (
+	"testing"
+)
+
+func TestSplitQuoted(t *testing.T) {
+	testCases := []struct {
+		input string
+		want  []string
+	}{
+		// All tests split on ; and treat ' as quoting character.
+		{
+			input: "",
+			want:  []string{""},
+		},
+		{
+			input: ";",
+			want:  []string{"", ""},
+		},
+		{
+			input: "'",
+			want:  []string{"'"},
+		},
+		{
+			input: "a;b",
+			want:  []string{"a", "b"},
+		},
+		{
+			input: "a;b;",
+			want:  []string{"a", "b", ""},
+		},
+		{
+			input: "a;b;c",
+			want:  []string{"a", "b", "c"},
+		},
+		{
+			input: "a;'b;c';d",
+			want:  []string{"a", "'b;c'", "d"},
+		},
+		{
+			input: "a;'b;c;d",
+			want:  []string{"a", "'b;c;d"},
+		},
```

Comment:
I see you're thinking here (treat unterminated quote as if terminated at EOF), but I'm not sure if it's the right approach. I don't necessarily think it's the wrong approach, either. I'm just not sure.

Reply:
I can picture two possible alternatives; if we hit the end of the string and find it has an unterminated quote:

1. Start over and split the entire string without respecting separator quoting at all -- aka how enmime worked previously, since it didn't have a notion of quoting.
2. Reset to the last opening quote and pretend it didn't exist; so earlier quoted runs would be respected.

I'd lean towards the first option, given we don't know which quote(s) may be missing, the second isn't going to be any safer than what the code does now.

Reply:
I think I'd agree with you about going with the first option.